### ISSUE TITLE
allow access to faster digitalWrite()

### ISSUE THREAD

**Issue by bxparks** [bxparks/AceSegment#1]
The `digitalWrite()` method on Arduino is known to be slow because the code needs to be able to handle pin numbers which are defined at runtime, see [discussions here](https://forum.arduino.cc/index.php/topic,4324.0.html). We can probably make `Driver::displayCurrentField()` a lot faster by allowing the segment pins to be a compile time constant and using a faster version of `digitalWrite()`.

**Comment by bxparks** [comment]
Brainstorming an idea (captured here so that I don't forget): `Hardware` could hold an array of pointers to member functions which implemented `digitalWrite01()`, `digitalWrite02()`,..., `digitalWriteXY()`, where XY is the largest expected pin number of a particular platform. Then `Hardware::writePin(pin)` could just do an index lookup of the array, and dispatch to that function.

**Comment by bxparks** [comment]
I tried implementing this at the `Hardware` layer, but the amount of CPU savings was relatively small, around 20 microseconds if I recall. 

Instead, I implemented this using code generation, through the `./tools/fast_driver.py` python script. This allows me to create `Driver` subclasses which bypass both the `Hardware` and `LedMatrix` abstraction layers, and write directly to the `digitalWriteFast()` methods. The reduction of CPU cycles is substantial, 60-120 microseconds, giving a `renderField()` that's 1.3 to 2.7 X faster.
